volatility3/framework/automagic/linux.py

```diff
@@ -184,6 +198,58 @@ def find_aslr(
         vollog.debug("Scanners could not determine any ASLR shifts, using 0 for both")
         return 0, 0
 
+    @classmethod
+    def find_aslr_vmcoreinfo(
+        cls,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        progress_callback: constants.ProgressCallback = None,
+    ) -> Optional[Tuple[int, int]]:
+        """Determines the ASLR offsets using the VMCOREINFO ELF note
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The layer within the context in which the module exists
+            progress_callback: A function that takes a percentage (and an optional description) that will be called periodically
+
+        Returns:
+            kaslr_shirt and aslr_shift
+        """
+
+        for (
+            _vmcoreinfo_offset,
+            vmcoreinfo,
+        ) in linux.VMCoreInfo.search_vmcoreinfo_elf_note(
```

Comment:
SInce this is versioned separately, we should check the version above (and bail gracefully) if it's different to what we expected.  That'll allow addition only bumps and major breaking bumps just fine.